lib/chat_api/reporting.ex

```diff
@@ -120,6 +121,18 @@ defmodule ChatApi.Reporting do
     end)
   end
 
+  @spec get_customer_breakdown(binary(), binary(), map()) :: [aggregate_by_field()]
+  def get_customer_breakdown(account_id, field, filters \\ %{}) do
+    Customer
+    |> where(account_id: ^account_id)
+    |> where(^filter_where(filters))
+    |> group_by([r], field(r, ^field))
+    # |> select([r], {field(r, ^field), count(r.id)})
+    |> select([r], %{:field => field(r, ^field), :count => count(r.id)})
+    |> order_by([r], desc: count(r.id))
+    |> Repo.all()
+  end
```

Comment:
@eikooc let's just do something like this for now :) 

```elixir
  @spec get_customer_breakdown(binary(), atom(), map()) :: [aggregate_by_field()]
  def get_customer_breakdown(account_id, field, filters \\ %{}) do
    Customer
    |> where(account_id: ^account_id)
    |> where(^filter_where(filters))
    |> group_by([r], field(r, ^field))
    |> select([r], {field(r, ^field), count(r.id)})
    |> order_by([r], desc: count(r.id))
    |> Repo.all()
    |> Enum.map(fn {value, count} -> %{field => value, :count => count} end)
  end
```

what do you think?

Reply:
That could definitely work. I'll look at it and update.

Reply:
We could also do `Enum.into(%{})` and get `%{"Chrome" => 1, "Firefox" => 1, "Safari" => 1}` as output. What do you think about that?

Reply:
hmm 🤔 I think I prefer the named properties for now!